client/js/views/LoginSignup.vue

```diff
@@ -126,24 +126,17 @@ export default {
     this.$store.dispatch('requestLogout');
   },
   methods: {
-    submitLogin() {
+    async submitLogin() {
       this.loading = true;
-      if (this.isLoggingIn) {
-        this.$store
-          .dispatch('requestLogin', {
-            username: this.username,
-            password: this.pw,
-          })
-          .then(this.handleLogin)
-          .catch(this.handleError);
-      } else {
-        this.$store
-          .dispatch('requestSignup', {
-            username: this.username,
-            password: this.pw,
-          })
-          .then(this.handleLogin)
-          .catch(this.handleError);
+      const action = this.isLoggingIn ? 'requestLogin' : 'requestSignup';
+      try {
+        await this.$store.dispatch(action, {
+          username: this.username,
+          password: this.pw,
+        });
+        this.handleLogin();
```

Comment:
I think there's fuckery here if the store actions don't return their respective promises, which is why I suggested the above changes. I would expect `this.handleLogin()` to fire without awaiting the store dispatch b/c of the missing return. I'm less familiar with `async/await` than with `.then().catch()` but it looks like we need that return in the store actions to me

Reply:
@itsalaidbacklife from my understanding, `this.handleLogin` will not fire until the dispatch above, and all `await`ed child calls _within_ the dispatch, wrap up execution.

As long as we are `await`ing all async calls within the dispatch:
```
// LoginSignup.vue
await this.$store.dispatch(requestLogin, {...}) 

// store/modules/auth.js (requestSignup())
await handleLogin(context, username, password, true);
...

// store/modules/auth.js (handleLogin())
const response = await fetch(`/user/${authType}`, {...})
...
return
```

then no returns should be necessary here with async/await structure! Since all async child calls from the dispatch are properly awaited, there should not be an issue with `this.handleLogin` firing prematurely

Reply:
Yep, this all tracks and I appreciate you all digging through this. We don't need these awaits as long as we're returning `this.handleLogin` since it's a promise, we're just passing it down the lane for the next bit of logic to handle. Good suggestions, thank you!

Reply:
I think you're right. My comment is now handled by [this commit: ](https://github.com/cuttle-cards/cuttle/pull/124/commits/7cc7ee9d92f8d95261c1583f7fe203f724dbd707)

This line didn't itself need to change, but we just had to ensure that the actions themselves returned correctly in order to correctly `await` them here